meshpy/space_time/beam_to_space_time.py

```diff
@@ -0,0 +1,378 @@
+
+        # Return the dat line.
+        return f'{self.i_global} {self.dat_pre_nodes} {self.four_c_name} {nodes_string} {self.dat_post_nodes} {" ".join(rotation_vectors)}'
+
+
+class SpaceTimeElementQuad4(SpaceTimeElement):
+    """A space-time element with 4 nodes."""
+
+    vtk_cell_type = vtk.vtkQuad
+    vtk_topology = list(range(4))
+    four_c_name = "QUAD4"
+
+
+class SpaceTimeElementQuad9(SpaceTimeElement):
+    """A space-time element with 9 nodes."""
+
+    vtk_cell_type = vtk.vtkQuadraticQuad
+    vtk_topology = list(range(9))
+    four_c_name = "QUAD9"
+
+
+def beam_to_space_time(
+    mesh_space_or_generator: Union[Mesh, Callable[[float], Mesh]],
+    time_duration: float,
+    number_of_elements_in_time: int,
+    *,
+    time_start: float = 0.0,
+) -> Tuple[InputFile, GeometryName]:
+    """Convert a MeshPy beam mesh to a surface space-time mesh.
+
+    Args
+    ----
+    mesh_space_or_generator:
+        Either a fixed spatial Mesh object or a function that returns the
+        spatial mesh for a given time. If this is a generator, the topology
+        of the mesh at the initial time is chosen for all times, only the
+        positions and rotations are updated.
+    time_duration:
+        Total time increment to be solved with the space-time mesh
+    number_of_elements_in_time:
+        Number of elements in time direction
+    time_start:
+        Starting time for the space-time mesh. Can be used to create time slaps.
+    Return
+    ----
+    space_time_mesh: InputFile
+        The space time mesh. Be aware that translating / rotating this mesh
+        might lead to unexpected results.
+        TODO: Return this as mesh. For now InputFile is better for performance reasons
+    return_set:
+        The nodes sets to be returned for the space time mesh:
+            "start", "end", "left", "right", "surface"
+    """
+
+    # Get the "reference" spatial mesh
+    if callable(mesh_space_or_generator):
+        mesh_space_reference = mesh_space_or_generator(time_start)
+    else:
+        mesh_space_reference = mesh_space_or_generator
+
+    # Perform some sanity checks
+    valid_element_types = [Beam3rLine2Line2, Beam3rHerm2Line3]
+    element_types = list(
+        set([type(element) for element in mesh_space_reference.elements])
+    )
+    if not len(element_types) == 1:
+        raise ValueError(
+            f"Expected all elements to be of the same type, got {element_types}"
+        )
+    element_type = element_types[0]
+    if element_type not in valid_element_types:
+        raise ValueError(f"Got unexpected element type {element_type}")
+
+    # Calculate global mesh properties
+    number_of_nodes_in_space = len(mesh_space_reference.nodes)
+    number_of_elements_in_space = len(mesh_space_reference.elements)
+    is_linear = element_type == Beam3rLine2Line2
+    space_time_element: Union[Type[SpaceTimeElementQuad4], Type[SpaceTimeElementQuad9]]
+    if is_linear:
```

Comment:
You could simply use
```suggestion
    if element_type == Beam2rLine2Line2:
```

here. This way the hard to read variable `is_linear` is gone and one directly sees what's happening here

You can also then use the other element in the else branch and raise an error if the element is another type

Reply:
Good point, I restructured the way types are defined in the space time function, now everything should hopefully be much clearer.